dockstore-client/src/test/java/io/dockstore/client/cli/WesToilIT.java

```diff
@@ -0,0 +1,155 @@
+     */
+    private boolean waitForWorkflowState(String runId, String state) throws InterruptedException {
+        for (int i = 0; i < WAIT_COUNT; i++) {
+
+            // Check on a workflow's run status
+            String[] commandStatementStatus = new String[]{ "workflow", "wes", "status",
+                "--config", TOIL_CONFIG,
+                "--id", runId
+            };
+            Client.main(commandStatementStatus);
+
+            // wait for the provided state to be printed
+            if (systemOutRule.getLog().contains(state)) {
+                return true;
+            }
+
+            Thread.sleep(WAIT_ITER_TIME_MILLI);
+        }
+
+        return false;
+    }
+
+    @Test
+    public void testBasicLaunch1() throws InterruptedException {
+        String[] commandStatementRun = new String[]{ "workflow", "wes", "launch",
+            "--config", TOIL_CONFIG,
+            "--entry", "github.com/dockstore-testing/wes-testing/single-descriptor-with-input:main",
+            "--json", ResourceHelpers.resourceFilePath("wesIt/w1_test.json"),
+            "--inline-workflow"
+        };
+        Client.main(commandStatementRun);
+        assertTrue("A helper command should be printed to stdout when a workflow is successfully started", systemOutRule.getLog().contains("dockstore workflow wes status --id"));
+
+        final String runId = findWorkflowId(systemOutRule.getLog());
+        final boolean isSuccessful = waitForWorkflowState(runId, COMPLETED_STATE);
+
+        if (!isSuccessful) {
+            fail("The workflow did not succeed in time.");
+        }
+    }
+
+    @Test
+    public void testBasicLaunch2() throws InterruptedException {
+        String[] commandStatementRun = new String[]{ "workflow", "wes", "launch",
+            "--config", TOIL_CONFIG,
+            "--entry", "github.com/dockstore-testing/wes-testing/multi-descriptor-no-input:main",
+            "--json", ResourceHelpers.resourceFilePath("wesIt/w1_test.json"), // Toil requires workflow_params to be provided
+            "--inline-workflow"
+        };
+        Client.main(commandStatementRun);
+        assertTrue("A helper command should be printed to stdout when a workflow is successfully started", systemOutRule.getLog().contains("dockstore workflow wes status --id"));
+
+        final String runId = findWorkflowId(systemOutRule.getLog());
+        final boolean isError = waitForWorkflowState(runId, EXECUTOR_ERROR_STATE);
+
+        // Toil does not support imports at this time, so multi-descriptor workflows should fail.
+        if (!isError) {
+            fail("The workflow was not supposed to succeed");
+        }
+    }
+
+    @Test
+    public void testBasicLaunch3() throws InterruptedException {
+        String[] commandStatementRun = new String[]{ "workflow", "wes", "launch",
+            "--config", TOIL_CONFIG,
+            "--entry", "github.com/dockstore-testing/wes-testing/single-descriptor-no-input:main",
```

Comment:
You could have a tag if you think main will ever change.

Reply:
Ideally, I think main shouldn't change to help with test maintenance. Added a note to the test repo's README to that effect: https://github.com/dockstore-testing/wes-testing/blob/main/README.md

Feel free to push back if there is any disagreement.